File: fd-player/src/examples/ghost_bicyle_js/bicycle/drive.js
return (frontCenter, rearCenter, frontRadius, frontTeeth, ratio, pedalAngle) => {
  const pedalLength = frontRadius * 2.5;
  const chainOffsetRatio = 0.4;
  const chainOffset = frontRadius * chainOffsetRatio;
  const rearChainOffset = frontRadius * ratio * chainOffsetRatio;

  const gear1 = lib.gear(frontCenter, frontRadius, frontTeeth, pedalAngle);
  const gear2 = lib.gear(rearCenter, frontRadius * ratio, frontTeeth * ratio, pedalAngle / ratio);

  const chain1 = chain(
    [frontCenter[0], frontCenter[1] + frontRadius + chainOffset],
    [rearCenter[0], rearCenter[1] + frontRadius * ratio + rearChainOffset],
    10,
    -pedalAngle * frontRadius
  );
  const chain2 = chain(
    [frontCenter[0], frontCenter[1] - (frontRadius + chainOffset)],
    [rearCenter[0], rearCenter[1] - (frontRadius * ratio + rearChainOffset)],
    10,
    pedalAngle * frontRadius
  );

  const pedal1 = cycle.pedal(frontCenter, pedalAngle, pedalLength, true);
  const pedal2 = cycle.pedal(frontCenter, pedalAngle + Math.PI, pedalLength, false);

  return {
    chain1,
    chain2,
    pedal1,
    gear1,
    pedal2,
    gear2
  };
};

File: examples/testlib/art/cartoon/transport/bicycle/drive.fs
(frontCenter, rearCenter, frontRadius, frontTeeth, ratio, pedalAngle) =>
{
  pedalLength: frontRadius * 2.5;
  pedalFootThickness: 0.6;
  pedalFootCenterRadius: pedalLength - pedalFootThickness / 2;
  chainOffsetRatio: 0.4;
  chainOffset: frontRadius * chainOffsetRatio;
  rearChainOffset: (frontRadius * ratio) * chainOffsetRatio;

  rearTeethRaw: frontTeeth * ratio;
  rearTeeth: math.Max(3, math.Floor(rearTeethRaw));

  gear1: cartoon.machine.parts.gear(frontCenter, frontRadius, frontTeeth, pedalAngle);
  gear2: cartoon.machine.parts.gear(rearCenter, frontRadius * ratio, rearTeeth, pedalAngle / ratio);

  chain1:
    chain(
      [frontCenter[0], frontCenter[1] + frontRadius + chainOffset],
      [rearCenter[0], rearCenter[1] + (frontRadius * ratio) + rearChainOffset],
      10,
      -pedalAngle * frontRadius);

  chain2:
    chain(
      [frontCenter[0], frontCenter[1] - (frontRadius + chainOffset)],
      [rearCenter[0], rearCenter[1] - ((frontRadius * ratio) + rearChainOffset)],
      10,
      pedalAngle * frontRadius);

  pedal1: pedal(frontCenter, pedalAngle, pedalLength, true);
  pedal2: pedal(frontCenter, pedalAngle + math.Pi, pedalLength, false);
  pedal1Center:
  [
    frontCenter[0] + math.Sin(pedalAngle) * pedalFootCenterRadius,
    frontCenter[1] + math.Cos(pedalAngle) * pedalFootCenterRadius
  ];
  pedal2Center:
  [
    frontCenter[0] + math.Sin(pedalAngle + math.Pi) * pedalFootCenterRadius,
    frontCenter[1] + math.Cos(pedalAngle + math.Pi) * pedalFootCenterRadius
  ];

  eval
  {
    chain1;
    chain2;
    pedal1;
    pedal1Center;
    gear1;
    pedal2;
    pedal2Center;
    gear2;
  };
}
